ClinGen/clincoded
# Connects to #782. VCI Basic Info tab updated changes.

This PR consists of changes pertinent to the Basic Info tab in addressing the following:
* Remove IDs (e.g. ClinVar Variation ID, dbSNP ID) from top of basic information tab since they are already displayed in the static header.
* Change the title of the first table from "Primary Transcript" to "ClinVar Primary Transcript".
* If a variant is not in ClinVar then put "No data was found for this allele in ClinVar. Search ClinVar for this variant." in this table.
* For now, add a table with a list of non NC_ hgvs names obtained from RefSeq if the variant is in ClinVar.
* Comment out "non-functional" Uniprot ID at the bottom of the tab temporarily.
* In the Ensembl transcript table, use the GRCh38 hgvs notation in the API call instead of the rsID to only obtain the relevant allele transcripts.
* Suppress the display of respective LinkOut links at the bottom of tab if either a variant is not in Clinvar or no data can be retrieved from Ensembl.

**Known issues:**
* When using Ensembl VEP api, only GRCh38 hgvs notation can return results.
* Tab's bottom LinkOut links are dependent on various pieces of obtained data from APIs to construct the URLs.
* When starting with a CA ID in the variant selection, choosing certain CD ID (e.g. "CA501068") does not return the expected GRCh38 hgvs name, although the GRCh37 hgvs name is in the response. As a result, there won't be any data displayed in the Ensembl table.

**Technical details:**
* Added mappings of GRCh38's NC_ genomic to chromosome to the mapping JSON file.
* Updated code to account for the absence of variant in ClinVar.

**Steps to test this PR:**
1. Login and proceed to the variant selection interface.
2. Expect the "pertinent" changes described above to be verifiable.
3. Try the following CA IDs to observe the different display of data: CA016023, CA501068, CA501058.
* CA016023 - returns both GRCh37 and GRCh38 and thus all 3 tables are expected to have data displayed.
* CA501068 - returns only GRCh37 and the variant does not exist in ClinVar. Thus none of the tables are expected to have data displayed.
* CA501058 - returns only GRCh38 and the variant does not exist in ClinVar. Thus only the "Ensembl Transcripts" table is expected to have data displayed.
